### ISSUE TITLE
Booking Period Restrictions

### ISSUE THREAD

**Issue by EvanDime** [CSU-Booking-Platform/application#18]
### User story
As a user, I cannot make a reservation if the current date does not fall within the booking period for that room. 

### Narrative
As a user, it shouldn't be possible to reserve things too early, as this would allow people to just over book everything months in advance. For example, the Mezzanine, cannot be booked more than fourteen days in advance. 

As an administrator, I must be able to configure a room's booking period. For example, Room A must be booked 10 business days in advance, but cannot be booked more than 2 months in advance. 

### Out of scope
Restrictions unrelated to the booking window

### Acceptance criteria

| # | Given | When | Then |
|---|-------|------|------|
|1|I am able to book rooms|I attempt to make a reservation and the current date is within the booking window|The system makes the reservation|
|2|I am able to book rooms|I attempt to make a reservation and the current date is not within the booking window|The system doe not make the reservation|

### UI Mockup:
![Booking Restriction](https://user-images.githubusercontent.com/31771444/94384386-79d46800-0110-11eb-9e2d-6e72a968ac26.PNG)

**Comment by Alex-Hayeur** [comment]
I've made a DB migration to add the necessary backend, once book a room is done I can add the logic for restricting a booking based off of the room's maximum booking in advance.

A few things for the feature to be ironed out in tomorrow's meeting though: Do rooms have a minimum days in advance to book, and will this check only be done when making a booking properly. I'd imagine it should be possible for an admin to force a booking into the system that ignores these kinds of restrictions.
